src/routes/guides/state-management.mdx

```diff
@@ -184,44 +179,95 @@ function Counter() {
 }
 ```
 
-While this approach works, it can be inefficient as the `doubleCount` signal will be re-evaluated on every render.
-When a function is computationally expensive, or used multiple times within a component, this can lead to performance issues.
-To avoid this, Solid introduced [Memos](/concepts/derived-values/memos):
+While this approach works for simple use cases, if `doubleCount` is used several times within a component or contains a computationally expensive calculation, it can lead to performance issues.
 
-```jsx
-import { createSignal, createEffect, createMemo } from "solid-js"
+The derived signal would be re-evaluated not just each time `count` is changed, but also for each use of `doubleCount()`.
+
+```jsx del={10} ins={11-14, 20-21}
+import { createSignal } from "solid-js"
 
 function Counter() {
 	const [count, setCount] = createSignal(0)
-	const [doubleCount, setDoubleCount] = createSignal(0)
 
-	// Memo
-	const squaredCount = createMemo(() => count() * count())
+	const increment = () => {
+		setCount(count() + 1)
```

Comment:
Can you please change the instances of this pattern on this file to be
```js
setCount(prev => prev + 1)
```
When doing the former in a memo/effect it will cause loops.